clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1908,24 +1908,25 @@ public void processConsumerRecord(
     /*
      * Report ingestion throughput metric based on the store version
      */
-    versionedIngestionStats.recordBytesConsumed(storeName, versionNumber, recordSize);
-    versionedIngestionStats.recordRecordsConsumed(storeName, versionNumber);
+    if (!record.getKey().isControlMessage()) { // skip control messages
```

Comment:
Why does it matter to skip control messages?

Reply:
Its not that critical, but it will actually diverge the record and byte consumed metrics. For control message byte size is returned as 0 and we still record them as consumed record.